calicoctl/commands/clientmgr/client.go

```diff
@@ -30,6 +31,24 @@ import (
 // parameter not loaded from file.
 func NewClient(cf string) (client.Interface, error) {
 	cfg, err := LoadClientConfig(cf)
+	// If user is not using calico specific configurations fallback to Kubernetes.
+	if cfg.Spec.EtcdEndpoints == "" && cfg.Spec.DatastoreType != apiconfig.Kubernetes {
```

Comment:
Alternate approach could be
```
if cfg.Spec.EtcdEndpoints == "" && cfg.Spec.DatastoreType == apiconfig.EtcdV3 || (cfg.Spec.DatastoreType == apiconfig.Kubernetes && cfg.Spec.Kubeconfig == "" && cfg.Spec.K8sAPIEndpoint == "")
```
@robbrockbank @caseydavenport  WDYT?

Reply:
I think we should change this in libcalico-go.  I'd suggest that we default to etcd if any of the other etcd parameters are explicitly set, otherwise default to kubernetes.

Reply:
Yeah, I think I agree. Ideally all of our clients that consume this sort of config behave consistently (and libcalico-go helps with that).